### ISSUE TITLE
diffFilter reporting incorrect errors for lines

### ISSUE THREAD

**Issue by alpipego** [exussum12/coverageChecker#63]
We have the coverage checker setup in a pre-commit hook. `phpcs.json` gets created only for changed files:

<details>
  <summary>relevant part of `pre-commit`</summary>

```php
define('GIT_ROOT', trim(shell_exec('git rev-parse --show-toplevel')));
define('PHPCS_BIN', sprintf('%1$s%2$svendor%2$sbin%2$sphpcs', GIT_ROOT, DIRECTORY_SEPARATOR));
define('PHPCBF_BIN', sprintf('%1$s%2$svendor%2$sbin%2$sphpcbf', GIT_ROOT, DIRECTORY_SEPARATOR));

// check if there are files to commit.
$files = [];
exec(sprintf('git diff --diff-filter=d --cached --name-only -- "*.php" ":(exclude)vendor%s*"', DIRECTORY_SEPARATOR), $files);
if (empty($files)) {
    exit();
}

echo 'Generating Diff...' . Colors::color('OK', 'green') . "\n";
shell_exec(sprintf('git diff --diff-filter=d --cached -- "*.php" ":(exclude)vendor%s*" > .phpcs-diff.txt', DIRECTORY_SEPARATOR));

echo 'Running PHP_CodeSniffer...';
// phpcs:ignore Generic.Files.LineLength.TooLong
$p    = popen(PHPCS_BIN . ' --cache  -pv --report=json --report-file=".phpcs.json" ' . implode(
        ' ',
        $files
    ) . ' || true', 'rb');
$last = microtime(true);
while (!feof($p)) {
    fgets($p);
    if ($last + .750 < microtime(true)) {
        echo '.';
        $last = microtime(true);
    }
}
flush();
pclose($p);
echo Colors::color('OK', 'green') . "\n";

echo 'Comparing Diffs...';
$diff = [];
exec(sprintf('vendor%1$sbin%1$sdiffFilter --phpcs .phpcs-diff.txt .phpcs.json 100', DIRECTORY_SEPARATOR), $diff);

$phpcs_errors = (empty($diff) || '100.00% Covered' === $diff[0]) ? 0 : 1;
if (0 !== $phpcs_errors) {
    echo Colors::color('failed', 'red') . "\n\n";
    echo Colors::color('Please fix these errors before committing:', 'black', 'red') . "\n\n";
    echo implode("\n", $diff);
} else {
    echo Colors::color('OK', 'green');
}
echo "\n";

exit($phpcs_errors);
```
</details>

<details>
<summary>part of diff</summary>

```
diff --git a/admin/admin.php b/admin/admin.php
index 936c67c..51e0ec3 100755
--- a/admin/admin.php
+++ b/admin/admin.php
@@ -724,8 +724,8 @@ class Advanced_Ads_Tracking_Admin {
             'advanced_ads_tracking_setting_section'
         );
 
-		if ($this->show_stats()) {
-			// open database management field
+		if ( $this->show_stats() ) {
+			// open database management field.
 			add_settings_field(
 				'tracking-db-mgmt',
 				__( 'Database Management', 'advanced-ads-tracking' ),
@@ -734,7 +734,7 @@ class Advanced_Ads_Tracking_Admin {
 				$advanced_section
 			);
 
-			// timeout for impressions sum transient
+			// timeout for impressions sum transient.
 			add_settings_field(
 				'sum-timeout',
 				__( 'Recalculate sums', 'advanced-ads-tracking' ),
@@ -742,6 +742,15 @@ class Advanced_Ads_Tracking_Admin {
 				$hook,
 				$advanced_section
 			);
+
+			// tracking for bots.
+			add_settings_field(
+				'tracking-bots',
+				__( 'Track bots', 'advanced-ads-tracking' ),
+				array( $this, 'render_settings_track_bots' ),
+				$hook,
+				$advanced_section
+			);
 		}
 
         // link base for public stats
@@ -753,15 +762,6 @@ class Advanced_Ads_Tracking_Admin {
             'advanced_ads_tracking_setting_section'
         );
 
-        // tracking for bots
-        add_settings_field(
-            'tracking-bots',
-            __( 'Track bots', 'advanced-ads-tracking' ),
-            array( $this, 'render_settings_track_bots' ),
-            $hook,
-            $advanced_section
-        );
```
</details>

<details>
<summary>Extract of `phpcs.json`</summary>

```
{
  "totals": {
    "errors": 2972,
    "warnings": 514,
    "fixable": 2857
  },
  "files": {
    "...\/admin\/admin.php": {
      "errors": 2149,
      "warnings": 198,
      "messages": [
        // more errors
        {
          "message": "Parameter comment must end with a full stop",
          "source": "Squiz.Commenting.FunctionComment.ParamCommentFullStop",
          "severity": 5,
          "fixable": false,
          "type": "ERROR",
          "line": 624,
          "column": 8
        },
        {
          "message": "Tabs must be used to indent lines; spaces are not allowed",
          "source": "Generic.WhiteSpace.DisallowSpaceIndent.SpacesUsed",
          "severity": 5,
          "fixable": true,
          "type": "ERROR",
          "line": 625,
          "column": 1
        },
        // more errors
        {
          "message": "Missing space before array closer.",
          "source": "WordPress.Arrays.ArrayDeclarationSpacing.NoSpaceBeforeArrayCloser",
          "severity": 5,
          "fixable": true,
          "type": "ERROR",
          "line": 722,
          "column": 66
        },
        {
          "message": "Tabs must be used to indent lines; spaces are not allowed",
          "source": "Generic.WhiteSpace.DisallowSpaceIndent.SpacesUsed",
          "severity": 5,
          "fixable": true,
          "type": "ERROR",
          "line": 723,
          "column": 1
        },
        {
          "message": "Tabs must be used to indent lines; spaces are not allowed",
          "source": "Generic.WhiteSpace.DisallowSpaceIndent.SpacesUsed",
          "severity": 5,
          "fixable": true,
          "type": "ERROR",
          "line": 724,
          "column": 1
        },
        {
          "message": "Tabs must be used to indent lines; spaces are not allowed",
          "source": "Generic.WhiteSpace.DisallowSpaceIndent.SpacesUsed",
          "severity": 5,
          "fixable": true,
          "type": "ERROR",
          "line": 725,
          "column": 1
        },
        {
          "message": "Tabs must be used to indent lines; spaces are not allowed",
          "source": "Generic.WhiteSpace.DisallowSpaceIndent.SpacesUsed",
          "severity": 5,
          "fixable": true,
          "type": "ERROR",
          "line": 756,
          "column": 1
        },
        {
          "message": "Tabs must be used to indent lines; spaces are not allowed",
          "source": "Generic.WhiteSpace.DisallowSpaceIndent.SpacesUsed",
          "severity": 5,
          "fixable": true,
          "type": "ERROR",
          "line": 757,
          "column": 1
        },
        {
          "message": "Tabs must be used to indent lines; spaces are not allowed",
          "source": "Generic.WhiteSpace.DisallowSpaceIndent.SpacesUsed",
          "severity": 5,
          "fixable": true,
          "type": "ERROR",
          "line": 758,
          "column": 1
        },
        {
          "message": "Tabs must be used to indent lines; spaces are not allowed",
          "source": "Generic.WhiteSpace.DisallowSpaceIndent.SpacesUsed",
          "severity": 5,
          "fixable": true,
          "type": "ERROR",
          "line": 759,
          "column": 1
        },
        {
          "message": "Tabs must be used to indent lines; spaces are not allowed",
          "source": "Generic.WhiteSpace.DisallowSpaceIndent.SpacesUsed",
          "severity": 5,
          "fixable": true,
          "type": "ERROR",
          "line": 760,
          "column": 1
        },
        // more errors
      ]
}
```
</details>

This is the output: 
![image](https://user-images.githubusercontent.com/2003667/87037328-51615f80-c1ec-11ea-865d-a615b199a103.png)

It seems that the diff hangs on some errors and keeps on reporting that error for multiple changed lines, even though these changed lines don't have an error and specifically not that error.

I'm looking forward to any hints on why this is happening and how to prevent it.

**Comment by alpipego** [comment]
Just noticed: The reported lines are within a method and the `Parameter comment must end with a full stop` error is on the DocBlock for that method (line 624). If I fix that error however line 624 now becomes a changed line that gets included in the diff and has more errors on it. 
Is it possible to prevent the outer errors from being reported on the changed lines?

**Comment by exussum12** [comment]
Its likely your diff is not quite right, Could you share the diff.txt and phpcs.json ? This usually happens when the diff doesn't accurately reflect whats happened

**Comment by alpipego** [comment]
Here's a gist with the diff and the json: https://gist.github.com/alpipego/5d9a7f343d44b03e20b291eb79a044d9

**Comment by exussum12** [comment]
Ok so I found the issue here, phpcs reports issues in places where they occur in most cases. In this case with the function comments, you can trigger them by editing the function itself. Eg editing the function causes the function comment to become outdated, and therefore needs to be flagged.

In this case that can't really happen, the CS failure relates to the comment itself, ill make an exception for this rule so it doesnt trigger when you just edit the function body

**Comment by exussum12** [comment]
Sorry I forgot to close this a release was made for this, try the latest version and this should work now

**Comment by alpipego** [comment]
I have updated to version 1.0.1, unfortunately, I don't see a change in the behavior. The error still gets reported for each changed line instead of just where it should be reported.

![image](https://user-images.githubusercontent.com/2003667/147488902-3022fad1-fc92-4bf5-a381-0544f3986c5b.png)
